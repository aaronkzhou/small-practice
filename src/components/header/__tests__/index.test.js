import React from 'react'
import TestRenderer from 'react-test-renderer'

describe('Header', () => {
    let Header

    beforeAll(() => {
        jest.doMock('react-router-dom')
        Header = jest.requireActual('../').default
    })

    afterAll(() => {
        jest.dontMock('react-router-dom')
    })

    it('header should render properly for `/` link, then the icon link should display as required', () => {
        const renderer = TestRenderer.create(<Header title={'Menu'} />)
        expect(renderer.toJSON()).toMatchSnapshot()
    })

    it('header should render properly for `/register-card` link, then the icon link should display as required', () => {
        const renderer = TestRenderer.create(<Header title={'register-card'} />)
        expect(renderer.toJSON()).toMatchSnapshot()
    })
})