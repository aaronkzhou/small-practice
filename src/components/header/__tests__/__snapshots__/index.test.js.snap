// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header should render properly for \`/\` link, then the icon link should display as required 1`] = `
<div
  className="header-wrapper"
>
  <span
    className="icon-wrapper"
  >
    <react-router-dom-link
      to="/register-card"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 176H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 272H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 368H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
        />
      </svg>
    </react-router-dom-link>
  </span>
  <span
    className="header-title"
  >
    Menu
  </span>
</div>
`;

exports[`Header header should render properly for \`/register-card\` link, then the icon link should display as required 1`] = `
<div
  className="header-wrapper"
>
  <span
    className="icon-wrapper"
  >
    <react-router-dom-link
      to="/"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
        />
      </svg>
    </react-router-dom-link>
  </span>
  <span
    className="header-title"
  >
    register-card
  </span>
</div>
`;
